projects/collection-editor-library/src/lib/components/fancy-tree/fancy-tree.component.ts

```diff
@@ -503,12 +506,111 @@ export class FancyTreeComponent implements OnInit, AfterViewInit, OnDestroy {
 
   moveDependentNodes(targetNode, currentNode) {
     const currentNodeDependency = this.editorService.getDependentNodes(currentNode.otherNode.data.id);
+    let nodeArray = [];
+    if(!_.isUndefined(currentNodeDependency)){
+    let nodeId = !_.isEmpty(currentNodeDependency.source) ? _.get(currentNodeDependency, 'source[0]') : _.get(currentNode, 'otherNode.data.id');
     if (!_.isEmpty(currentNodeDependency.target)) {
-      _.forEach(currentNodeDependency.target, id => {
+      if (currentNode.hitMode === 'after') {
+        const dependentNode = this.treeService.getNodeById(nodeId);
+        dependentNode.moveTo(targetNode, currentNode.hitMode);
+        _.forEach(currentNodeDependency.target, id => {
+          const dependentNode = this.treeService.getNodeById(id);
+          dependentNode.moveTo(currentNode.otherNode, currentNode.hitMode);
+        });
+      }
+      else {
+        nodeArray.push(nodeId); 
+       _.forEach(currentNodeDependency.target,id => {
+         nodeArray.push(id);
+       });
+       _.forEach(nodeArray, id => {
         const dependentNode = this.treeService.getNodeById(id);
         dependentNode.moveTo(targetNode, currentNode.hitMode);
       });
+      }
+    }
+    this.updateBranchingLogic(_.get(targetNode,'data.metadata.parent'),nodeId);
+   }
+  }
+
+  updateBranchingLogic(targetId,nodeId){
+    console.log(targetId, nodeId);
+    const data = this.treeService.getFirstChild();
+    let hierarchyData = this.editorService._toFlatObj(data);
+    console.log(hierarchyData)
+    let sectionList=[];
+    this.branchingObject={}
+    this.treeData = Object.keys(hierarchyData);
+    _.forEach(this.treeData, id => {
+      if (hierarchyData[id].root === true) {
+        sectionList = hierarchyData[id].children
+        console.log(sectionList);
+      }
+      if (sectionList.includes(id)) {
+        let data = this.editorService.getBranchingLogicByFolder(id);
+        let MyArray = Object.keys(data);
+        if(hierarchyData[id].children.length != MyArray.length){
+          this.createBranchinglogic(hierarchyData,id,data);
+        }
+      }
+    });
+  }
+
+  createBranchinglogic(hierarchyData, nodeId, data) {
+    let children = hierarchyData[nodeId].children;
+    let sectionName = hierarchyData[nodeId].name;
+    let dataArray=Object.keys(data);
+    console.log("createBranchinglogic");
```

Comment:
keep the method names consistent (always camelCase) . and start using tslint/eslint so that we have consistent spaces, semicolons, brackets and structure to the code.